# Geometric Distribution

There are 1,000,000 users and we want to expose 1,000 users per day to a test. The same user can be selected twice for the test.

1. What's the expected value of how many days someone needs to wait before they receive the test?

The probability of being selected for a user is,

$$
\frac{1}{1,000,000}
$$

The probability of not being selected for a user is,

$$
1 - \frac{1}{1,000,000}
$$

But the 1,000 selections are performed per day, so the real probability of not being selected by 1,000 selections is,

$$
(1 - \frac{1}{1,000,000})^{1,000}
$$

In [1]:
(1 - (1 / 1000000)) ** 1000

0.9990004993338457

We have 99.9% probability that a user will not be selected on a given day. So the real probability of being selected by exposing 1,000 users is,

$$
1 - (1 - \frac{1}{1,000,000})^{1,000}
$$

In [2]:
1 - (1 - (1 / 1000000)) ** 1000

0.000999500666154285

With a certain probability, we wanna know the number of days before being selected. So we can use **geometric distribution** if we think of the number of days as trials before being selected as success. The expected value of geometric distribution with a probability $p$ is,

$$
E(X) = \frac{1}{p}
$$

In this question, $p = 1 - (1 - \frac{1}{1,000,000})^{1,000}$. The expected value is,

$$
\frac{1}{1 - (1 - \frac{1}{1,000,000})^{1,000}}
$$

In [4]:
1 / (1 - (1 - (1 / 1000000)) ** 1000)

1000.4995833045679

So the expected value of the number of days ti wait before being selected for the test is 1000,5.

2. What is the probability they get selected after the first day? Is it closer to 0 or 1?

The probability of being selected on the first day is, as we computed above,

$$
1 - (1 - \frac{1}{1,000,000})^{1,000}
$$
$$
\approx 0.0009995
$$
$$
\approx 0.001
$$

So the probability of being selected after the first day is the negation of the probability so,

$$
1 - \left[ 1 - (1 - \frac{1}{1,000,000})^{1,000} \right]
$$

In [7]:
1 - (1 - (1 - (1 / 1000000))**1000)

0.9990004993338457

The probability of being selected after the first day is 99.9%, so it's closer to 1.

## Resource

- [Geometric distribution](https://en.wikipedia.org/wiki/Geometric_distribution)